src/bridge/cockpitrouter.c

```diff
@@ -1631,6 +1665,102 @@ cockpit_router_dbus_startup (CockpitRouter *router)
     }
 }
 
+/* Superuser init */
+
+static void superuser_init_step (CockpitRouter *router);
+
+static void
+superuser_init_start (CockpitRouter *router,
+                      const gchar *id,
+                      const gchar *password)
+{
+  router->superuser_init_in_progress = TRUE;
+  router->superuser_init_next_rule = router->rules;
+  router->superuser_init_id = g_strdup (id);
+  router->superuser_init_password = g_strdup (password);
```

Comment:
We are rather careless about the password. This of course already applies to the current implementation, but let's try better. We keep the password in cleartext in an unprivileged process, and copy it around *a lot*. As a first step, could superuser_init() not dup it, but pass it to this function as non-const, and `router` takes ownership of it? Let's please also mlock() the password as soon as we get it, so that it doesn't get swapped onto the disk in clear.

Reply:
Yes... Pasing the password in the "init" message is not so great either.  It used to be in "authorize" messages, but it is arguably worse to suddenly turn the well know "init" message into a carrier of secrets. I have a look at the new memfd infrastructure that we have.

Reply:
Ok, I have added a WIP commit that uses the old `authorize` messages to transport the password.  That is not great, but at least not worse that what we have been living with all these years.

Reply:
We go with this approach.